src/main/java/entities/Timeslot.java

```diff
@@ -9,12 +9,24 @@
     private String room;
     private LocalTime start;
     private LocalTime end;
+    private char session;
 
-    public Timeslot(LocalTime start, LocalTime end, DayOfWeek day, String room) {
+    /**
+     * @param start Start of the timeslot
+     * @param end End of the timeslot
+     * @param day Day the timeslot occurs
+     * @param room Room the room of the timeslot
+     * @param session The session this timeslot occurs in (F/S). Note that this is different from a
+     *     course's session, as the timeslot can only represent one "half" of a Y course's timeslot
+     */
+    public Timeslot(LocalTime start, LocalTime end, DayOfWeek day, String room, char session) {
         this.start = start;
         this.end = end;
         this.day = day;
         this.room = room;
+        this.session = session;
```

Comment:
So basically, since we are now internally storing `Timeslots` for Y Courses by splitting them into a `S` Timeslot and a `F` timeslot, that means the definition of Session is different for a `Course` vs. a `Timeslot`, since a `Timeslot` session can't be 'Y'. I think that's confusing, and ideally fixed by a better name, but I can't think of one.

Reply:
maybe something like semesterSession not sure though